cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/templates/slurm_parallelcluster_slurmdbd.conf

```diff
@@ -1,12 +1,17 @@
-# slurm_parallelcluster_slurmdbd.conf is managed by the pcluster processes.
-# Do not modify.
-# Please add user-specific slurmdbd configuration options in slurmdbd.conf
-{% if scaling_config.Database.Uri is defined %}
-DbdHost={{ head_node_config.head_node_hostname }}
-StorageHost={{ scaling_config.Database.Uri | uri_host }}
-StoragePort={{ scaling_config.Database.Uri | uri_port }}
-{# Dashes in StorageLoc cause issues with the database creation #}
-StorageLoc={{ cluster_name | replace("-", "_") }}
-StorageUser={{ scaling_config.Database.UserName }}
-StoragePass=dummy
-{% endif %}
+{# Adding comments at the beginning of each line is a trick to indent the template without affecting the output #}
```

Comment:
I think we generate the configuration files using the `trim_blocks=True` and `lstrip_blocks=True` which should let us use indentation in the template without it appearing in the generated file.

Do the indentations appear in the generated file?

Reply:
I had written a nice comment that got lost :/

Long story short, those options only remove the new lines and spaces caused by the presence of the `{# ... #}` Jinja blocks, not any indentation we may want to apply to the rest of the template to make the logic more understandable. Meaning that yes, if I indent lines that do not have Jinja blocks in them, they will remain indented in the rendered output.

I couldn't find a nicer alternative to this.

My impression is that this trick makes the logic embedded in the template more easy to follow, although it becomes slightly uglier to read. I would be in favor of keeping this trick.